src/test_l3_dsVideoDevice.c

```diff
@@ -0,0 +1,659 @@
+*
+*/
+void test_l3_dsVideoDevice_Term()
+{
+    gTestID = 11;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsVideoDevice_Term()");
+    status = dsVideoDeviceTerm();
+    UT_LOG_INFO("Result dsVideoDeviceTerm() OUT:dsError_t=[%s]",  UT_Control_GetMapString(dsErrorMappingTable, status));
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief Set Zoom mode of Source device
+*
+* This test sets zoom mode of video device seected.
+* **Test Group ID:** 03@n
+* **Test Case ID:** 002@n
+* **Pre-Conditions:** None@n
+* VideoDevice_init Should be called before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select SetZoomMode from list
+*
+*/
+
+void test_l3_dsVideoDevice_SetZoomMode()
+{
+    dsError_t status   = dsERR_NONE;
+    gTestID = 2;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+    int32_t choice,j;
+
+    dsVideoDevice_getHandle();
+    UT_LOG_MENU_INFO(" \t  Supported Zoom Modes are:");
+    for (j = 0; j <= dsVIDEO_ZOOM_MAX; j++)
+    {
+        UT_LOG_MENU_INFO("\t%d.  %-20s ", j, \
+                    UT_Control_GetMapString(dsVideoZoomMappingTable, j));
+    }
+
+    UT_LOG_MENU_INFO("------------------------------------------");
+    UT_LOG_MENU_INFO(" Select the Zoom mode:");
+    scanf("%d", &choice);
+    readAndDiscardRestOfLine(stdin);
+
+    UT_LOG_INFO("Calling dsSetDFC(IN:Handle:[0x%0X],IN:dsVideoZoom_t:[%s])",gHandle, \
+                    UT_Control_GetMapString(dsVideoZoomMappingTable, choice));
+    status = dsSetDFC(gHandle, choice);
+    UT_LOG_INFO("Result dsSetDFC(), OUT:dsError_t=[%s]", UT_Control_GetMapString(dsErrorMappingTable, status));
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief Set Device Frame Rate of Sink device
+*
+* This test sets frame rate of video device selected.
+* **Test Group ID:** 03@n
+* **Test Case ID:** 003@n
+* **Pre-Conditions:** None@n
+* VideoDevice_init Should be called before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select SetDisplayFrameRate from list
+*
+*/
+void test_l3_dsVideoDevice_SetDisplayFramerate()
+{
+    gTestID = 3;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+    dsError_t status   = dsERR_NONE;
+    int32_t choice,j;
+
+    dsVideoDevice_getHandle();
+    UT_LOG_MENU_INFO(" \t  Supported Display Framerate are:");
+    for (j = 0; j <= dsVIDEO_FRAMERATE_MAX; j++)
+    {
+        UT_LOG_MENU_INFO("\t%d.  %-20s ", j, UT_Control_GetMapString(dsVideoFrameRateTable, j));
+    }
+
+    UT_LOG_MENU_INFO("------------------------------------------");
+    UT_LOG_MENU_INFO(" Select the Display Framerate :");
+    scanf("%d", &choice);
```

Comment:
please validate the input and `please take care in other functions also `

Reply:
Updated as per request

Reply:
in few cases missed [] for out prams